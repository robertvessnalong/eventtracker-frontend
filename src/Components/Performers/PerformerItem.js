import { Card, Button, Col } from 'react-bootstrap';
import { Link } from 'react-router-dom';

const PerformerItem = ({ performer }) => {
  return (
    <Card className='mt-4 mb-4 p-0 text-white'>
      <Card.Img src={performer.images.huge} alt='Card image' />
      <Card.ImgOverlay className='center-content'>
        <Card.Title style={{ fontWeight: 'bolder' }} as='h2'>
          {performer.name}
        </Card.Title>
        <Col xs={12} md={6}>
          <Link to={`/performers/${performer.id}`} rel='noreferrer'>
            <Button className='blue-btn' variant='primary'>
              See Events
            </Button>
          </Link>
        </Col>
      </Card.ImgOverlay>
    </Card>
  );
};

export default PerformerItem;
