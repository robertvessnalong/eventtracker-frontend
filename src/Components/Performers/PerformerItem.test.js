import { render, screen, waitFor, fireEvent } from '@testing-library/react';
import PerformerItem from './PerformerItem';
import TestUser from '../../Testing/TestUser';
import TestPerformer from '../../Testing/TestPerfomer';
import UserContext from '../../Context/UserContext';
import { BrowserRouter } from 'react-router-dom';

describe('Events Component', () => {
  // Smoke Test
  test('renders without crashing', async () => {
    render(
      <BrowserRouter>
        <UserContext.Provider value={{ user: TestUser }}>
          <PerformerItem
            performer={TestPerformer}
            favorites={TestUser.favorites}
          />
        </UserContext.Provider>
      </BrowserRouter>
    );
    await waitFor(() => {
      const headerSpan = screen.getByText(/Billy Joel/i);
      expect(headerSpan).toBeInTheDocument();
    });
  });

  //Snapshot Test
  test('matches snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <UserContext.Provider value={{ user: TestUser }}>
          <PerformerItem
            performer={TestPerformer}
            favorites={TestUser.favorites}
          />
        </UserContext.Provider>
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
