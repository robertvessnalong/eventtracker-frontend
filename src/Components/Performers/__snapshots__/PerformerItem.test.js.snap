// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="mt-4 mb-4 p-0 text-white card"
  >
    <img
      alt="Card image"
      class="card-img"
      src="https://seatgeek.com/images/performers-landscape/billy-joel-82da62/303/huge.jpg"
    />
    <div
      class="center-content card-img-overlay"
    >
      <h2
        class="card-title"
        style="font-weight: bolder;"
      >
        Billy Joel
      </h2>
      <div
        class="col-md-6 col-12"
      >
        <a
          href="/performers/303"
          rel="noreferrer"
        >
          <button
            class="blue-btn btn btn-primary"
            type="button"
          >
            See Events
          </button>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
